sdk/typespec/typespec_client_core/src/http/policies/retry/mod.rs

```diff
@@ -22,36 +35,35 @@ type DateTimeFn = fn() -> OffsetDateTime;
 
 /// Get the duration to delay between retry attempts, provided by the headers from the response.
 ///
-/// This function checks for retry-after headers in the following order, following the
-/// JS Azure SDK implementation: <https://github.com/Azure/azure-sdk-for-js/blob/17de1a2b7f3ad61f34ff62876eced7d077c10d4b/sdk/core/core-rest-pipeline/src/retryStrategies/throttlingRetryStrategy.ts#L35>
-/// * `retry-after-ms`
-/// * `x-ms-retry-after-ms`
-/// * `Retry-After`
+/// This function checks for retry-after headers in the following order:
+///
+/// 1. (TODO) `retry-after-ms`
+/// 2. (TODO) `x-ms-retry-after-ms`
+/// 3. `Retry-After`
 ///
-/// If no header is provided, None is returned.
-pub(crate) fn get_retry_after(headers: &Headers, datetime_now: DateTimeFn) -> Option<Duration> {
-    [RETRY_AFTER_MS, X_MS_RETRY_AFTER_MS, RETRY_AFTER]
-        .iter()
-        .find_map(|header| {
-            headers.get_str(header).ok().and_then(|v| {
-                if header == &RETRY_AFTER {
-                    // RETRY_AFTER values are either in seconds or a HTTP date
-                    v.parse::<u64>().ok().map(Duration::from_secs).or_else(|| {
-                        try_parse_retry_after_http_date(v).map(|retry_after_datetime| {
-                            let now = datetime_now();
-                            if retry_after_datetime < now {
-                                Duration::from_secs(0)
-                            } else {
-                                date::diff(retry_after_datetime, now)
-                            }
-                        })
+/// If no header is provided, `None` is returned.
+pub fn get_retry_after(headers: &Headers, now: DateTimeFn) -> Option<Duration> {
+    // TODO: Need to figure out a way to pass these in - perhaps in `RetryOptions`?
+    [RETRY_AFTER].iter().find_map(|header| {
```

Comment:
I was thinking that last night as well. I did actually move (hmm...did I move it back for now?) `retry-after-ms` to `typespec_client_core` at least since it's not obviously Microsoft-specific, but it's also not standard, strangly. I'd think ms precision in this day and age might be better in some cases...though, maybe HTTP isn't the protocol for that either.

I'll "hook" this somehow in another PR. I was considering either a callback of sorts, or just something we can attach to the `RetryOptions` like a `Vec<(HeaderName, FnMut<???>)>` or something. But, like you also suggest, that feels like overkill.

The Go runtime that @gracewilcox has been working on effectively overrides this but Go is duck-typed, unlike Rust. So they have an easier time replacing policies and options all up. Lucky dogs...err, gophers.

Reply:
Opened #1753 to track.